device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -338,110 +338,121 @@ void openLinks()
         }
     }
 
-    boolean sendMessage(Message message)
+    SendResult sendMessage(Message message)
     {
         if (this.deviceClientConfig.getDeviceId().equals(message.getConnectionDeviceId()))
         {
-            if (message.getMessageType() == null)
-            {
-                message.setMessageType(DEVICE_TELEMETRY);
-            }
+            return SendResult.WRONG_DEVICE;
```

Comment:
This is a frequently executed line of code that it would be a nuisance to log here. The AMQP connection is iterating over its sessions to find the one corresponding to this device. For instance, if a multiplexed connection had 500 devices, each telemetry message sent out could hit this line of code up to 499 times. I can certainly add some comments around what this means, though

Reply:
What does that mean about logging then? Still spurious and unhelpful?

Reply:
It turns this line of code into a "Should never happen" type scenario. I'll add a log for it now